nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -80,6 +85,35 @@ in {
 
   options.services.nextcloud = {
     enable = mkEnableOption (lib.mdDoc "nextcloud");
+
+    enableBrokenCiphersForSSE = mkOption {
+      type = types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        This option uses OpenSSL PHP extension linked against OpenSSL 1.x rather
+        than latest OpenSSL (≥ 3), this is not recommended except if you need
+        it.
+
+        Server-side encryption in NextCloud uses RC4 ciphers, a broken cipher
+        since ~2004.
+
+        This cipher has been disabled in OpenSSL ≥ 3 and requires
+        a specific legacy profile to re-enable it.
+
+        If you upgrade to a NextCloud using OpenSSL ≥ 3 and have
+        server-side encryption configured, you will not be able to access
+        your files anymore, enabling this option can restore access to your files.
+
+        It is advised to [disable server-side encryption](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/encryption_configuration.html#disabling-encryption) as it is unclear
+        it provides any amount of security beyond encryption for S3-style buckets
```

Comment:
I'd probably say "external storage" rather than S3 as the reasoning applies to any kind of storage you're not operating on your own. Also, I'd probably alter the wording a bit here: "Unless you're using external storage, it's advised to [disable server-side encryption](...)`.

Reply:
Done.